substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -2473,6 +2513,844 @@ static boolean not(byte b) {
+                // not reachable
+                //  1910      } else {
+            } else {
+                //  1911          fd = (*env)->GetIntField(env, fdObj, IO_fd_fdID);
+                fd = PosixUtils.getFD(fdObj);
+                //  1912      }
+            }
+            //  1913      if (IS_NULL(iaObj)) {
+            //  1914          JNU_ThrowNullPointerException(env, "iaObj");
+            //  1915          return;
+            //  1916      }
+            Objects.requireNonNull(iaObj, "iaObj");
+            //  1917
+            //  1918      /*
+            //  1919       * Determine if this is an IPv4 or IPv6 join/leave.
+            //  1920       */
+            //  1921      ipv6_join_leave = ipv6_available();
+            ipv6_join_leave = JavaNetNetUtil.ipv6_available();
+            //  1922
+            //  1923  #ifdef __linux__ /* elided */
+            //  1929  #endif
+            //  1930
+            //  1931      /*
+            //  1932       * For IPv4 join use IP_ADD_MEMBERSHIP/IP_DROP_MEMBERSHIP socket option
+            //  1933       *
+            //  1934       * On Linux if IPv4 or IPv6 use IP_ADD_MEMBERSHIP/IP_DROP_MEMBERSHIP
+            //  1935       */
+            //  1936      if (!ipv6_join_leave) {
+            if (! ipv6_join_leave) {
+                //  1937  #ifdef __linux__ /* elided */
+                //  1939  #else
+                //  1940          struct ip_mreq mname;
+                NetinetIn.ip_mreq mname = StackValue.get(NetinetIn.ip_mreq.class);
+                //  1941  #endif
+                //  1942          int mname_len;
+                int mname_len = 0;
+                //  1943
+                //  1944          /*
+                //  1945           * joinGroup(InetAddress, NetworkInterface) implementation :-
+                //  1946           *
+                //  1947           * Linux/IPv6:  use ip_mreqn structure populated with multicast
+                //  1948           *              address and interface index.
+                //  1949           *
+                //  1950           * IPv4:        use ip_mreq structure populated with multicast
+                //  1951           *              address and first address obtained from
+                //  1952           *              NetworkInterface
+                //  1953           */
+                //  1954          if (niObj != NULL) {
+                if (niObj != null) {
+                    //  1955  #if defined(__linux__) /* elided */
+                    //  1972  #endif
+                    //  1973              {
+                    {
+                        //  1974                  jobjectArray addrArray = (*env)->GetObjectField(env, niObj, ni_addrsID);
+                        final Enumeration<InetAddress> addresses = niObj.getInetAddresses();
+                        //  1975                  jobject addr;
+                        InetAddress addr;
```

Comment:
[The value of the local variable addr is not used](https://travis-ci.org/oracle/graal/jobs/548381854#L2740).

Reply:
This one was a real bug!